### PR TITLE
doc: Zero downtime deployments document

### PR DESCRIPTION
**Pull request by draychev**
Adding a document outlining a method to decrease 502 errors during rolling updates.

### CODE REVIEW

File: docs/zero-downtime-deployments.md [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+    spec:
+      containers:
+      - name: ctr
+        ...
+      terminationGracePeriodSeconds: 101
+```
+
+4. Decrease interval between App Gateway health probes to backend pools. The goal is to increase number of probes per unit of time. This will ensure that a terminated pod, which has not yet been removed from App Gateway's backend pool, will be marked as unhealthy sooner, thus removing the probability of a request landing on a terminated pod and resulting in a 502 error.
+
+
```
Comment (akshaysngupta) [inline docs/zero-downtime-deployments.md]:
```yaml
readinessProbe:
  httpGet:
    path: /
    port: 80
  periodSeconds: 5
  timeoutSeconds: 5
  failureThreshold: 3
```

File: docs/zero-downtime-deployments.md [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+          preStop:
+            exec:
+              command: ["sleep","90"]
+```
+
+
+> Note: The "sleep" command assumes the container is based on Linux. For Windows containers the equivalent command is "timeout".
+
+The addition of the `preStop` container life cycle hook will:
+  - delay Kubernetes sending `SIGTERM` to the container by 90 seconds, but put the pod immediately in `Terminating` state
```
Comment (asridharan) [inline docs/zero-downtime-deployments.md]: This is just a thought: This is pretty specific to Linux containers? Not sure if there is a solution for Windows containers and we need to explicitly highlight this fact?

**Comment by akshaysngupta** [comment]
Also, Can you move this into how-tos folder and add a link to this on the root readme ?

**Comment by draychev** [comment]
Renamed doc to `Minimizing Downtime During Deployments`
@akshaysngupta PTAL

**Pull request merged**